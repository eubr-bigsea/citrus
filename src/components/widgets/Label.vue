<template>
    <span class="label">
        <span :data-order="field.order" :data-name="field.name"
            :class="{'text-danger': error, required: field.required}" :title="error">{{field.label || field.name}}</span>
        <span v-show="field.required">*</span>:
        <span v-if="(showHelp === undefined || showHelp) && field.help" class="data-help">
            <span class="fa fa-question-circle float-right" :title="field.help"></span>
        </span>
    </span>
</template>

<script>
    import Widget from '../../mixins/Widget.js';
    export default {
        mixins: [Widget],
        name: "property-label",
        props: {
            field: { required: true, type: Object },
            value: {},
            showHelp: true,
        },
        computed: {
            error() {
                if (this.field.required && (this.value === undefined || this.value === null || this.value === '' || this.value === {})) {
                    return this.$tc('errors.missingRequiredValue');
                }
                return null;
            },
        }
    }
</script>
<style scoped>
    .label {
        font-size: 1em;
    }
</style>
