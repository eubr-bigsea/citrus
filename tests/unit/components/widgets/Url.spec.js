import { shallowMount } from '@vue/test-utils';
import UrlComponent from '$SRC/components/widgets/Url.vue';
import Label from '$SRC/components/widgets/Label.vue';
import { expect, it } from 'vitest';

describe('Url.vue', () => {
    it('read-only is true and value is null', () => {
        const wrapper = shallowMount(UrlComponent, {
            propsData: {
                field: {
                    value: null,
                    default: 'https://www.google.com.br/'
                },
                readOnly: true,
                value: null
            }
        });

        expect(wrapper.findComponent(Label).exists()).toBe(false);
        expect(wrapper.text()).toBe("https://www.google.com.br/");

    });
    
    it('read-only is false and value is correct', () => {
        const wrapper = shallowMount(UrlComponent, {
            propsData: {
                field: {
                    value: 'https://www.google.com.br/',
                    default: 0
                },
                readOnly: false,
                value: null
            }
        });

        expect(wrapper.findComponent(Label).exists()).toBeTruthy();
        const urlarea = wrapper.find('urlarea');
    });

    it('snapshots the component', () => {
        const wrapper = shallowMount(UrlComponent, { propsData: { field: {} } });
        expect(wrapper.text()).toMatchSnapshot();
    });
});