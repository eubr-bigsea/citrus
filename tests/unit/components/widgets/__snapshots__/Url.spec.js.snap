// Vitest Snapshot v1

exports[`Url.vue > snapshots the component 1`] = `""`;
